File: app/src/main/kotlin/com/spacex/ui/rocketDetails/RocketDetailsUIMDecorator.java
package com.spacex.ui.rocketDetails;

import android.content.Context;
import android.graphics.drawable.Drawable;
import androidx.annotation.NonNull;
import androidx.core.content.ContextCompat;

public class RocketDetailsUIMDecorator {

  private final Context context;
  private RocketDetailsUIM mRocketDetailsUIM;

  RocketDetailsUIMDecorator(@NonNull final Context context,
      @NonNull final RocketDetailsUIM rocketDetailsUIM) {
    this.context = context;
    this.mRocketDetailsUIM = rocketDetailsUIM;
  }

  public Drawable getImage() {
    return ContextCompat.getDrawable(context, mRocketDetailsUIM.getImageResId());
  }

  public String getName() {
    return mRocketDetailsUIM.getName();
  }

  public String getDescription() {
    return mRocketDetailsUIM.getDescription();
  }
}

File: app/src/main/kotlin/com/spacex/ui/rocketDetails/RocketDetailsUIMDecorator.kt
package com.spacex.ui.rocketDetails

import android.content.Context
import android.graphics.drawable.Drawable
import androidx.core.content.ContextCompat

internal class RocketDetailsUIMDecorator(
        private val context: Context, private val rocketDetailsUIM: RocketDetailsUIM) {

    val image: Drawable?
        get() = ContextCompat.getDrawable(context, rocketDetailsUIM.imageResId)

    val name: String
        get() = rocketDetailsUIM.name

    val description: String
        get() = rocketDetailsUIM.description
}
